app/models/solidus_stripe/gateway.rb

```diff
@@ -56,6 +94,35 @@ def capture(amount_in_cents, payment_intent_id, options = {})
       )
     end
 
+    # Authorizes and captures a certain amount on the provided payment source.
+    #
+    # See `#authorize` for how the confirmation step is performed.
+    #
+    # @todo add support for purchasing custom amounts
+    def purchase(amount_in_cents, _source, options = {})
+      check_given_amount_matches_payment_intent(amount_in_cents, options)
+
+      stripe_payment_intent_options = {
+        amount: to_stripe_amount(amount_in_cents, options[:originator].currency),
+        confirm: false,
+        capture_method: "automatic",
+      }
+
+      stripe_payment_intent = SolidusStripe::PaymentIntent.prepare_for_payment(
+        options[:originator],
+        **stripe_payment_intent_options,
+      )
+
+      confirm_stripe_payment_intent(stripe_payment_intent.stripe_intent_id)
+
+      build_payment_log(
+        success: true,
+        message: "PaymentIntent was confirmed and captured successfully",
+        response_code: stripe_payment_intent.id,
+        data: stripe_payment_intent.stripe_intent,
+      )
```

Comment:
Should we add error handling here as well?

Reply:
🙏 ✔️ 
https://github.com/solidusio/solidus_stripe/blob/140bfbcb23c0bb8a4a693d9cb6b050056cd19958/app/models/solidus_stripe/gateway.rb#L124-L129

Reply:
Should we add tests?